Fails to compile on SPARC due to missing CHAR_CORENAME

After fixing #1442, compiling on SPARC still fails:
```
openblas_get_config.c:58:3: error: expected ',' or ';' before 'CHAR_CORENAME'
   CHAR_CORENAME
   ^
openblas_get_config.c: In function 'openblas_get_corename':
openblas_get_config.c:81:10: error: 'CHAR_CORENAME' undeclared (first use in this function)
   return CHAR_CORENAME;
          ^
openblas_get_config.c:81:10: note: each undeclared identifier is reported only once for each function it appears in
openblas_get_config.c:85:1: warning: control reaches end of non-void function [-Wreturn-type]
 }
 ^
make[3]: *** [Makefile:113: openblas_get_config.o] Error 1
```

The problem is that the autogenerated `config.h` does not define `CHAR_CORENAME`:
```
#define OS_SUNOS        1
#define ARCH_SPARC      1
#define C_GCC   1
#define __64BIT__       1
#define PTHREAD_CREATE_FUNC     pthread_create
#define BUNDERSCORE     _
#define NEEDBUNDERSCORE 1
#define SPARC
#define V9
#define DTB_DEFAULT_ENTRIES 32
#define GEMM_MULTITHREAD_THRESHOLD      4
```

When passing `TARGET=SPARC`, everything works because a line `#define CHAR_CORENAME "sparc"` is added.

This should be easy to fix analogous to #1443 but I don't know what the right fix is. Should `CHAR_CORENAME` be defined or not?

--- a/cpuid_sparc.c
+++ b/cpuid_sparc.c
@@ -57,3 +57,8 @@ void get_cpuconfig(void){
 void get_libname(void){
   printf("v9\n");
 }
+
+void get_corename(void){
+  printf("sparc\n");
+}
+

--- a/cpuid_sparc.c
+++ b/cpuid_sparc.c
@@ -58,7 +58,7 @@ void get_libname(void){
   printf("v9\n");
 }
 
-void get_corename(void){
-  printf("sparc\n");
+char *get_corename(void){
+  return "sparc";
 }
 

--- a/getarch.c
+++ b/getarch.c
@@ -1116,7 +1116,7 @@ int main(int argc, char *argv[]){
 #ifdef FORCE
     printf("CORE=%s\n", CORENAME);
 #else
-#if defined(INTEL_AMD) || defined(POWER) || defined(__mips__) || defined(__arm__) || defined(__aarch64__) || defined(ZARCH)
+#if defined(INTEL_AMD) || defined(POWER) || defined(__mips__) || defined(__arm__) || defined(__aarch64__) || defined(ZARCH) || defined(sparc)
     printf("CORE=%s\n", get_corename());
 #endif
 #endif
@@ -1224,7 +1224,7 @@ int main(int argc, char *argv[]){
 #ifdef FORCE
     printf("#define CHAR_CORENAME \"%s\"\n", CORENAME);
 #else
-#if defined(INTEL_AMD) || defined(POWER) || defined(__mips__) || defined(__arm__) || defined(__aarch64__) || defined(ZARCH)
+#if defined(INTEL_AMD) || defined(POWER) || defined(__mips__) || defined(__arm__) || defined(__aarch64__) || defined(ZARCH) || defined(sparc)
     printf("#define CHAR_CORENAME \"%s\"\n", get_corename());
 #endif
 #endif

